### DISCUSSION TITLE
Бэкапы и планы запросов

### DISCUSSION THREAD

**Discussion by pingvin1308** [cleannetcode/LessonMonitor#88] [MSSQL]
Ветка для выполнения заданий не потребуется. Работу следует размешать ниже в данной дискуссии.

https://www.sqlservertutorial.net/load-sample-database/
https://www.sqlservertutorial.net/sql-server-sample-database/

Почитать:
https://docs.microsoft.com/en-us/sql/relational-databases/performance/execution-plans?view=sql-server-ver15

Сделать:
Попробовать создать бэкап и восстановить базу. (результат - скрин где видно копию базы)
Попробовать сделать пару запросов и вывести их план (результат - скрин где видно планы запросов)
* сделать два запроса с одинаковым планом (результат - скрин где видно планы запросов)

Проверить:
провести ревью кода любого чужого пулл реквеста относящегося к данной домашке.

**Comment by AshtonBro** [comment]
### (MSSQL - 5) Бэкапы и планы запросов #88  <h3>
#### 1. Создан backup БД двума способами. <h4>

![image](https://user-images.githubusercontent.com/54149866/120382531-2ff5ee80-c33d-11eb-8f05-487662a858dd.png) #29 

* 1.1 Через **задачу** резервого копирования.
![image](https://user-images.githubusercontent.com/54149866/120382617-4ac86300-c33d-11eb-9d80-924c5dee5adf.png)

* 1.2 Через **экспорт** резервого копирования.
![image](https://user-images.githubusercontent.com/54149866/120382725-6d5a7c00-c33d-11eb-8bad-f5098ef5ca38.png)

Результат:
![image](https://user-images.githubusercontent.com/54149866/120382958-b6aacb80-c33d-11eb-8ece-d9438435cb7d.png)

#### 2. Восстановление БД двумя способами. <h4>

* 2.1 Через задачу **восстановить**.
![image](https://user-images.githubusercontent.com/54149866/120383123-e6f26a00-c33d-11eb-9ea4-bb85abee18bc.png)
![image](https://user-images.githubusercontent.com/54149866/120383138-ee197800-c33d-11eb-803f-2e533ac1ce5d.png)
![image](https://user-images.githubusercontent.com/54149866/120383152-f2de2c00-c33d-11eb-8bdc-03aba6ce7e6b.png)

* 2.2 Через **импорт** восстановления.
![image](https://user-images.githubusercontent.com/54149866/120383221-1012fa80-c33e-11eb-9407-e3784511f650.png)

#### 3. Вывод план запроса <h4>

![image](https://user-images.githubusercontent.com/54149866/120383413-481a3d80-c33e-11eb-9941-2916d4b6c7f9.png)

#### 4. **Задание со звёздочкой** - выполнены запросы с одинаковым планом. <h4>

* 4.1 План запроса 1
![image](https://user-images.githubusercontent.com/54149866/120384039-1655a680-c33f-11eb-8304-84dd7c9542a7.png)

* 4.2 План запроса 2
![image](https://user-images.githubusercontent.com/54149866/120384073-240b2c00-c33f-11eb-878f-eba6447d1281.png)

**Comment by Misha902** [comment]
сделал бэкап бд, первым способом

![1](https://user-images.githubusercontent.com/79799958/121706249-1c394d80-cade-11eb-9b5c-7addeb82ad45.png)
![2](https://user-images.githubusercontent.com/79799958/121706294-2b200000-cade-11eb-881f-925f4af4fd3e.png)
![3](https://user-images.githubusercontent.com/79799958/121706451-50ad0980-cade-11eb-99a9-0f8168abab49.png)

сделал восстановление бд
![4](https://user-images.githubusercontent.com/79799958/121706635-7cc88a80-cade-11eb-81e4-e7f4a529d080.png)
![5](https://user-images.githubusercontent.com/79799958/121706654-7fc37b00-cade-11eb-8f24-bed0099fe9f8.png)

Сделал пару запросов и получил их план
![1 1](https://user-images.githubusercontent.com/79799958/121706756-9964c280-cade-11eb-89d4-1cb58cf138b2.png)
![2 1](https://user-images.githubusercontent.com/79799958/121706775-9d90e000-cade-11eb-9e3b-3ba209bb6773.png)
![3 1](https://user-images.githubusercontent.com/79799958/121706795-a1246700-cade-11eb-9fa0-d078f75c733c.png)

**Reply by AshtonBro** [reply]
Не стал пробовать делать 2 запроса что бы получить один и тот же план ?

**Reply by Misha902** [reply]
Видел твой вариант, свой не придумал
